src/Rasterization/Renderers/SVGRectRenderer.php

```diff
@@ -21,25 +23,87 @@ protected function prepareRenderParams(SVGRasterizer $rasterizer, array $options
         $y1 = self::prepareLengthY($options['y'], $rasterizer) + $rasterizer->getOffsetY();
         $w  = self::prepareLengthX($options['width'], $rasterizer);
         $h  = self::prepareLengthY($options['height'], $rasterizer);
+        $rx = self::prepareLengthX($options['rx'], $rasterizer);
+        $ry = self::prepareLengthY($options['ry'], $rasterizer);
 
         return array(
             'x1' => $x1,
             'y1' => $y1,
             'x2' => $x1 + $w - 1,
             'y2' => $y1 + $h - 1,
+            'rx' => $rx - 1,
+            'ry' => $ry - 1,
         );
     }
 
     protected function renderFill($image, array $params, $color)
     {
+        $x1 = $params['x1'];
+        $y1 = $params['y1'];
+        $x2 = $params['x2'];
+        $y2 = $params['y2'];
+        $rx = $params['rx'];
+        $ry = $params['ry'];
+
+        if (($rx !== 0) || ($ry !== 0)) {
+            self::renderFillRounded($image, $params, $color);
+            return ;
+        }
         imagefilledrectangle(
             $image,
-            $params['x1'], $params['y1'],
-            $params['x2'], $params['y2'],
+            $x1, $y1,
+            $x2, $y2,
             $color
         );
     }
 
+    private function renderFillRounded($image, array $params, $color) {
+        $x1 = $params['x1'];
+        $y1 = $params['y1'];
+        $x2 = $params['x2'];
+        $y2 = $params['y2'];
+        $rx = $params['rx'];
+        $ry = $params['ry'];
+
+        imagefilledrectangle(
+            $image,
+            $x1 + $rx, $y1,
+            $x2 - $rx, $y2,
+            $color
+        );
+        imagefilledrectangle(
+            $image,
+            $x1, $y1 + $ry,
+            $x2, $y2 - $ry,
+            $color
+        );
```

Comment:
Drawing 2 overlapping rectangles is not a good idea. Think about colors with transparency. In those cases the part where they overlap would be colored differently than the rest — the same is true for the filled ellipses below. I think we need 3 rectangles, one in the middle, one on the left and one on the right (or middle+top+bottom). The ellipses need to be converted to arcs.